src/utils/Fetch.ts

```diff
@@ -67,11 +59,40 @@ const fetch = <T>(url: string): Promise<T> => {
     request.on('socket', socket => {
       socket.on('timeout', () => {
         request.abort();
-        reject(`Socket timeout while connecting to '${url}'`);
+        reject(`Socket timeout while connecting to '${options.host}${options.path}'`);
       });
       socket.setTimeout(SOCKET_TIMEOUT);
     });
-  });
+
+    request.end();
+  })
+}
+
+/**
+ * Fetch JSON data from a remote HTTP endpoint and de-serialize to the provided type.
+ * There are no guarantees the response will conform to the contract defined by T.
+ * It is up to the consumer to ensure the provided T captures all possible response types
+ * from the provided endpoint.
+ *
+ * @param url - currently only supports HTTP
+ */
+const fetch = async <T>(url: string): Promise<T> => {
```

Comment:
Can we delete this and change the call in ECSEnvironment to use fetchWithOptions instead?

We could also just simplify this class to `fetch` and `fetchString`, with both functions taking options as a parameter.